src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php

```diff
@@ -153,16 +153,10 @@ public function getKernel()
     /**
      * Gets the Surrogate instance.
      *
-     * @throws \LogicException
-     *
      * @return SurrogateInterface A Surrogate instance
      */
     public function getSurrogate()
     {
-        if (!$this->surrogate instanceof Esi) {
-            throw new \LogicException('This instance of HttpCache was not set up to use ESI as surrogate handler. You must overwrite and use createSurrogate');
```

Comment:
@Tobion, `createSurrogate` is here https://github.com/symfony/symfony/blob/master/src/Symfony/Bundle/FrameworkBundle/HttpCache/HttpCache.php#L73